### ISSUE TITLE
ROI detection issues in v0.6.3

### ISSUE THREAD

**Issue by tobiasackels** [MouseLand/suite2p#193]
Hi, 

after upgrading to v0.6.3 I can no longer reproduce the ROI detection that I got with v0.5.5. 
I changed parameters systematically (spatial_scale, threshold, max_overlap, preclassifiy) but without success.
Any suggestions on how to improve or get back to previous results would be greatly appreciated!

Thanks a lot in advance!
___________________________________________________________________________________________________

This is what I got with v0.5.5 after manual curation (threshold 1.5):
![Ver0 55_meanE](https://user-images.githubusercontent.com/36504295/59513222-4e455400-8eb2-11e9-9106-d31848d9a0db.PNG)

Here is a list of all the combinations I used with v0.6.3:
![image](https://user-images.githubusercontent.com/36504295/59513374-a3816580-8eb2-11e9-8522-24a5b4d2427b.png)

And here are some examples of the test runs with v0.6.3:

**RUN#1**
![image](https://user-images.githubusercontent.com/36504295/59513425-ba27bc80-8eb2-11e9-86e1-ce01ea43f9b3.png)

**RUN#10**
![image](https://user-images.githubusercontent.com/36504295/59513507-e2172000-8eb2-11e9-8047-7b49f4b58162.png)

**RUN#13**
![image](https://user-images.githubusercontent.com/36504295/59513550-f9eea400-8eb2-11e9-8182-6edec44efa5c.png)

**Comment by ransona** [comment]
I second that. I haven't played with the parameters as thoroughly as tobiasackels but pretty sure we're getting a lot less rois with a similar config to the version from around Feb / March. It looks like there are things in the correlation map which I would have thought would have been detected and do overlap with active cells like in the attached screen grab.
Thanks
Adam

![image](https://user-images.githubusercontent.com/7849066/59620190-26f5bd80-9124-11e9-8e5b-21f86087b289.png)

**Comment by carsen-stringer** [comment]
I think some things are getting thrown out by the classifier, but generally doesn't look great. The github version now runs the old cell detection by default (`ops['sparse_mode']=0`). I'll update the pip later today. In our recordings we do find more cells with `ops['sparse_mode']=1` but I think it needs more testing.

**Comment by carsen-stringer** [comment]
pip working (`pip install suite2p --upgrade`)

**Comment by ransona** [comment]
Hi @tobiasackels  - are you happy with how this is working now? i.e. is it more or less reproducing what you saw before the upgrade? I ask because I'm still not sure it is working as well as before on our data...

**Comment by chrytsi** [comment]
Hi all,
same for me: after upgrading I cannot reproduce my previous results when I re-run suite2p on the same set of data with exactly the same parameters as before. I now detected only 98 ROIS (vs. 144 before), and importantly, any detected ROIs now contain more pixels from the surrounding neuropil. The correlation map also looks comletely different.
I attach a figure showing the correlation map and detected ROIs for the same dataset in the old and new version:
![old_vs_new](https://user-images.githubusercontent.com/34349487/64442095-1de30f80-d0c7-11e9-8c0c-a0786abb1c4a.png)

Many thanks,
Chryssanthi

**Comment by carsen-stringer** [comment]
@chrysti I think in your case you might have run with ops['sparse_mode']=1 and then with it off? I did make a release in which it was on by default but then turned it off in later releases. The one on the left looks like ops['sparse_mode']=1 which I recommend for most data

**Comment by chrytsi** [comment]
@carsen-stringer I went back and ran this dataset in several versions of suite2p.
When I run this same dataset in suite2p 0.6.16 with ops['sparse_mode']=1 I only get **3** ROIs. 
In the above example the left has been run in v0.6.3, the right in 0.6.16 with ops['sparse_mode']=0.

I can replicate the bad result pictured on the right in every version from 0.6.7 onwards (v0.6.4, 0.6.5, and 0.6.6 are interrupted by errors. 
Long story short: the last version performing well on my ROI detection is v0.6.3.

**Comment by carsen-stringer** [comment]
There are two versions of the ROI detection: sparse_mode and sourcery. v0.6.3 only allowed sparse_mode to be on (if you ran from the GUI). In later versions I made sourcery the default (sparse_mode=0), but I still recommend using sparse_mode=1 for most cases. Therefore with v0.6.3 you were likely running sparse_mode but with a different threshold or spatial scale detected. It might have auto detected the spatial scale incorrectly in the newer version. You can force a certain spatial scale with spatial_scale (hover over it to see what the numbers mean or look at the documentation). Also I'd recommend decreasing the  "threshold_scaling" so that more ROIs are found.

Does the correlation map look the same in v0.6.3 and v0.6.16 sparse_mode=1?

**Comment by carsen-stringer** [comment]
I am going to set sparse_mode to default in the next release. If you are still having issues getting good results with sparse_mode you can try changing the binning by changing "tau" (the binning for the cell detection is the decay time in frames = ops['tau'] * ops['fs']). You can also try high-pass filtering your data before cell detection with the "high_pass" parameter. I'd also make sure that your registration is working well. If it's still not great, you can keep "sparse_mode" set to 0.
